velox/functions/lib/DateTimeFormatter.h

```diff
@@ -175,10 +175,22 @@ class DateTimeFormatter {
 
   DateTimeResult parse(const std::string_view& input) const;
 
+  /// Returns max size of the formatted string. Can be used to preallocate
+  /// memory before calling format() to avoid extra copy.
+  uint32_t maxResultSize(const date::time_zone* timezone) const;
+
   std::string format(
       const Timestamp& timestamp,
       const date::time_zone* timezone) const;
 
+  /// Result buffer is pre-allocated according to maxResultSize.
+  /// Returns actual size.
+  int32_t format(
```

Comment:
Is the other format method still used? Can we update callsites and remove it?

Reply:
Currently, this method is only used by DateTimeFormatterTest.cpp. Maybe, we can keep it here for possible use at other place.

Reply:
We don't want this method to be used since it is slow. Let's remove it.

Reply:
Just removed it in latest commit. Thanks!